core/controllers/editor_test.py

```diff
@@ -1220,3 +1224,125 @@ def test_email_functionality_cannot_be_used_by_non_moderators(self):
                 expected_status_int=401)
 
             self.logout()
+
+
+class EditorAutosaveTest(BaseEditorControllerTest):
+    """Test the handling of editor autosave actions."""
+
+    EXP_ID1 = '1'
+    EXP_ID2 = '2'
+    EXP_ID3 = '3'
+    NEWER_DATETIME = datetime.datetime.strptime('2017-03-16', '%Y-%m-%d')
+    OLDER_DATETIME = datetime.datetime.strptime('2015-03-16', '%Y-%m-%d')
+    DRAFT_CHANGELIST = [{
+        'cmd': 'add_state',
+        'state_name': 'State 5'}]
+    NEW_CHANGELIST = [{
+        'cmd': 'edit_exploration_property',
+        'property_name': 'title',
+        'new_value': 'New title'}]
+    INVALID_CHANGELIST = [{
+        'cmd': 'edit_exploration_property',
+        'property_name': 'title',
+        'new_value': 1}]
+
+    def setUp(self):
```

Comment:
I would try to move as much of this as possible into the individual functions so that it's easier to follow what's happening.